README.md

```diff
@@ -70,12 +70,10 @@ best way to work with the project.
 
 ## Running with Docker
 
-Start by pulling the developer image from the [GitHub Container Registry][ghcr-core-dev] and then start the developer shell:
+Start by running the developer shell:
 
 ```shell
-$ docker pull ghcr.io/dependabot/dependabot-core-development:latest
-$ docker tag ghcr.io/dependabot/dependabot-core-development dependabot/dependabot-core-development
```

Comment:
I'm very hesitant to remove these, even with the updater broken apart, it is still a _significant_ speedup to pull these rather than building locally, and most folks reading the Readme aren't going to be working in a codespace.

Why can't we instead recommend they pull `docker pull ghcr.io/dependabot/dependabot-core-development-<ecosystem>:latest`??

Reply:
We aren't currently publishing development images since they don't add much on top of the updater image they're built on, which is why I removed this.

We could suggest pulling the ecosystem image `ghcr.io/dependabot/dependabot-<ecosystem>` but... this README was written in simpler times when we all were running the same architecture. I think until we get an arm64 image published it would be unwise for folks running arm64 to run the amd64 image as emulation is quite slow. 

I could mention that caveat here.

Reply:
I ran the existing commands and I got an error:

`invalid argument "ghcr.io/dependabot/dependabot-updater-NPM" for "-t, --tag flag: invalid reference format: repository name must be lowercase`. Had errors with lowercase npm and yarn too. 

Trying go_modules now; these might be newbie mistakes so open to hearing more thougthts.

Reply:
@abdulapopoola it should be `bin/docker-dev-shell npm_and_yarn`

The ecosystem is the directory in this project. I can add a blurb about that here as well. @jeffwidman also created an issue to track fixing this confusion: https://github.com/dependabot/dependabot-core/issues/6577

Reply:
`go_modules` worked; so I guess it is a case of knowing the right params to pass? Is there a list somewher to smoothen the onboarding process for new devs?

Reply:
Yeah, I also documented this at `CONTRIBUTING.md` [recently](https://github.com/dependabot/dependabot-core/pull/6582), but the README still needed an update. I went with this to document how to get the name of the ecosystem:

> Then, assuming you're working on a single language, you'll need to start a development container for that language through
>
> ```
> bin/docker-dev-shell <ecosystem>
> ```
>
> The name of the ecosystem should be one of the top level root folders in this repo.

Reply:
I pushed an update that addresses this, let me know what you all think!